src/user/serializers.py

```diff
@@ -1086,6 +1087,9 @@ class Meta:
     def get_achievements(self, author):
         return author.achievements
 
+    def get_achievements_details(self, author):
```

Comment:
Given that we are trying to get out of performance hell, I think this pattern will hurt us.
Nothing wrong with the code you are writing here, but rather the pattern of how things are serialized is problematic (I think).

Explanation:
```
        upvote_count = getattr(self.user, "upvote_count", 0)
        peer_review_count = getattr(self.user, "peer_review_count", 0)
        amount_funded = getattr(self.user, "amount_funded", 0)
```

These lines are already invoked in the serializer in the `get_achievements` method. As a result, I think that we will double up our queries. 

This observation is not tested and it could be that Django is somehow smart enough to avoid doubling up the queries. Use the debug toolbar to verify. 

If it is indeed doubling up the queries, we should use another approach such as overriding the `to_representation` serializer method

Reply:
I have replaced the old get_achievements instead, this will avoid the double fetching.